opm/grid/cpgrid/processEclipseFormat.cpp

```diff
@@ -271,7 +279,14 @@ namespace cpgrid
 #ifdef VERBOSE
         std::cout << "Building geometry." << std::endl;
 #endif
-        buildGeom(output, cell_to_face_, cell_to_point_, face_to_output_face, geometry_.geomVector(std::integral_constant<int,0>()),
+        // here we need the cell volumes based on the active index order
+        std::unordered_map<size_t, double> aquifer_cell_volumes_local;
+        for (size_t i = 0; i < this->global_cell_.size(); ++i) {
+            if (aquifer_cell_volumes.count(this->global_cell_[i]) > 0) {
+                aquifer_cell_volumes_local.insert({i, aquifer_cell_volumes.at(this->global_cell_[i])});
+            }
```

Comment:
This will search the map twice for the same cell index.  I think I'd personally write this with iterators along the lines of
```C++
for (auto nc = this->global_cell_.size(), i = 0*nc; i < nc; ++i) {
    auto aquCellPos = aquifer_cell_volumes.find(this->global_cell_[i]);
    if (aquCellPos != aquifer_cell_volumes.end()) {
        aquifer_cell_volumes_local.emplace(i, aquCellPos->second);
    }
}
```

Reply:
Yes. You are right. Thanks for pointing it out.